optimum/intel/openvino/modeling_decoder.py

```diff
@@ -491,6 +495,7 @@ def forward(
             # the first condition at the function beginning above.
             # It should be something that is not None and it should be True when converted to Boolean.
             past_key_values = ((),)
+            self.past_len += input_ids.shape[1]
```

Comment:
I would prefer to have a method `_get_past_length` with `input_ids` and `past_key_values` arguments returning the expected past_length  instead of having a `past_len` attribute which needs to be updated at different steps (which will get complicated for maintenance in my opinion). WHat do you think @eaidova ?

Reply:
@echarlaix  I'm not sure that it is possible, past_len was introduced primarily for stateful model, because we can not touch past_key_values that are stored inside stateful model (it can become complicated and non-effective because the conception of stateful model provides managing memory and format for them on runtime device level, API that returns kv cache back to users will break it and requires to repack them in user-friendly format it can cost additional resources and time)

Reply:
@eaidova the following might help, you can find the current length in stateful models by using the following: 
```python
self.request.query_state()[0].state.shape[2]
```

Reply:
@ofirzaf i know how to query state, what I'm trying to say that we need to check size on each inference, it is not recommended to do it so often and will lead to perf and memory menegment issus due to packing/unpacking states, that is why using calculation of shapes without quiry state is preferred from openvino runtime perspective

Reply:
moved to usage function